tests/test_commands_task.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# # -*- coding: utf-8 -*-
+
+"""
+Unit tests for Task Commands.
+________________________________________________________________________________
+
+Created by brightSPARK Labs
+www.brightsparklabs.com
+"""
+
+# standard libraries
+import subprocess
+from pathlib import Path
+from time import sleep, time
+from typing import List
+
+import pytest
+from click.testing import CliRunner
+
+# vendor libraries
+from ruamel.yaml import YAML
+
+# local libraries
+from appcli.commands.configure_cli import ConfigureCli
+from appcli.commands.task_cli import TaskCli
+from appcli.configuration_manager import ConfigurationManager
+from appcli.logger import enable_debug_logging, logger
+from appcli.models.cli_context import CliContext
+from appcli.models.configuration import Configuration
+from appcli.orchestrators import DockerComposeOrchestrator
+
+# ------------------------------------------------------------------------------
+# CONSTANTS
+# ------------------------------------------------------------------------------
+
+APP_NAME = "TEST_APP"
+
+# directory containing this script
+BASE_DIR = Path(__file__).parent
+
+STACK_CONFIGURATION_FILE = Path(BASE_DIR, "resources/test_stack_settings.yml")
+
+DOCKER_COMPOSE_YML = Path(
+    BASE_DIR, "resources/templates/baseline/docker-compose.tasks.yml"
+)
+
+DOCKER_COMPOSE_SERVICES = list(YAML().load(open(DOCKER_COMPOSE_YML))["services"].keys())
+
+# ------------------------------------------------------------------------------
+# TASKS
+# ------------------------------------------------------------------------------
+
+
+class Test_TaskCommands:
+    def test_task_run_headless_arg_long(self, test_env):
+
+        TIME_START = time()
+        result = test_env.invoke_task_command(["run", "--detach", "sleep-1"])
+        TIME_END = time()
+
+        assert TIME_END - TIME_START < 0.1
```

Comment:
Using a sleep() command in this test and measuring that time difference is a **proxy** to test for the fact that the '-d/--detach' argument was passed to our subprocess.

I don't love this because it's:
- deterministic because your timer may run slow for whatever reason meaning that tests that should pass may fail for unknown reasons, and
- you could just test for the existence of the correct flag directly, which proves that the executed command is as expected.

I think you need to change the patch to return a boolean to say whether the task would have been run in 'detached' mode or not